### PR TITLE
Add support UIA provider for CheckedListBox control

### PR DESCRIPTION
**Pull request by DmitryGorokhov**
Related Issue: #3421

## Proposed changes

- Switched on the `SupportsUiaProviders` property of `CheckedListBox` class.
- Updated uia properties.
- Added unit tests.

## Customer Impact

- Improving development experience for CheckedListBox control accessibility.

## Regression? 

- No

## Risk

- Minimal

#### CheckedListBox parent under inspect

clb fw parent

#### CheckedListBox item under inspect

clb fw item checked

#### CheckedListBox control under narrator

nb cld before

#### CheckedListBox parent under inspect

new 22

#### CheckedListBox item under inspect

new 11

## Test methodology

- Manually.
- Unit tests.
- Automation tests.

## Accessibility testing

- Inspect
- Narrator
- Accessibility Insights

## Test environment(s) 

- .Net SDK version:  6.0.100-rc.1.21377.33

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5276)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.cs [hunk 1]
```diff
@@ -845,6 +845,13 @@ protected override void OnKeyPress(KeyPressEventArgs e)
         protected virtual void OnItemCheck(ItemCheckEventArgs ice)
         {
             _onItemCheck?.Invoke(this, ice);
+            if (IsAccessibilityObjectCreated)
+            {
+                AccessibleObject checkedItem = AccessibilityObject.GetChild(ice.Index);
+                object type = checkedItem.GetPropertyValue(UiaCore.UIA.ControlTypePropertyId);
+                checkedItem.RaiseAutomationPropertyChangedEvent(UiaCore.UIA.NamePropertyId, Name, Name);
+                checkedItem.RaiseAutomationPropertyChangedEvent(UiaCore.UIA.ControlTypePropertyId, type, type);
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.cs:853]:
```suggestion
                AccessibleObject checkedItem = AccessibilityObject.GetChild(ice.Index);
                checkedItem.RaiseAutomationPropertyChangedEvent(UiaCore.UIA.ToggleToggleStatePropertyId, ice.CurrentValue, ice.NewValue);
```

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs [hunk 2]
```diff
@@ -103,5 +103,76 @@ public void CheckedListBoxAccessibleObject_Role_IsExpected_ByDefault(bool create
...
             Assert.Equal(createControl, checkedListBox.IsHandleCreated);
         }
+
+        [WinFormsTheory]
+        [InlineData(0)]
+        [InlineData(3)]
+        public void CheckedListBoxAccessibleObject_GetChildCount_ReturnsExpected(int childCount)
+        {
+            using CheckedListBox checkedListBox = new();
+            for (int i = 0; i < childCount; i++)
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs:113]: Add a blank line before `for`

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 3]
```diff
@@ -0,0 +1,260 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:5]: Move this class upper to SWF folder, don't keep it in AccessibleObjects folder

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxAccessibleObject.cs [hunk 4]
```diff
@@ -2,109 +2,76 @@
 // The .NET Foundation licenses this file to you under the MIT license.
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxAccessibleObject.cs:2]: The class will be easier:
image

Think about how to avoid the type cast.

Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxAccessibleObject.cs:2]: Put the following to the `ListBoxAccessibleObject.GetChild` method:
image

**Review by vladimir-krestov — COMMENTED** [review]
Needs to be reworked

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs [hunk 5]
```diff
@@ -39,7 +39,7 @@ public void CheckedListBoxAccessibleObject_CheckBounds()
 
         [WinFormsTheory]
         [InlineData(true, (int)UiaCore.UIA.ListControlTypeId)]
-        [InlineData(false, (int)UiaCore.UIA.PaneControlTypeId)]
+        [InlineData(false, (int)UiaCore.UIA.ListControlTypeId)]
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs:42]: Is this behavior consistent with the old behavior?

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs [hunk 6]
```diff
@@ -103,5 +103,66 @@ public void CheckedListBoxAccessibleObject_Role_IsExpected_ByDefault(bool create
...
+        [WinFormsFact]
+        public void CheckedListBoxAccessibleObject_FragmentNavigate_NavigateToFirstChild_IsExpected()
+        {
+            using CheckedListBox checkedListBox = new();
+            AccessibleObject accessibleObject = checkedListBox.AccessibilityObject;
+
+            for (int i = 0; i < 3; i++)
+            {
+                checkedListBox.Items.Add(i);
+            }
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs:143]: I think in this case we do not need a `for (int i = 0; i < 3; i++)`. Please fix it in the other tests
```suggestion
            checkedListBox.Items.Add(0);
            checkedListBox.Items.Add(1);
            checkedListBox.Items.Add(2);
```

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs [hunk 7]
```diff
@@ -103,5 +103,66 @@ public void CheckedListBoxAccessibleObject_Role_IsExpected_ByDefault(bool create
...
+
+            AccessibleObject expected = accessibleObject.GetChild(0);
+
+            Assert.Equal(expected, accessibleObject.FragmentNavigate(UiaCore.NavigateDirection.FirstChild));
+            Assert.False(checkedListBox.IsHandleCreated);
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs:148]: We usually return `null` when the `Handle` is not created. @vladimir-krestov, could you please share your opinion?

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+using static System.Windows.Forms.CheckedListBox;
+using static Interop;
+
+namespace System.Windows.Forms.Tests.AccessibleObjects
+{
+    public class CheckedListBoxItemAccessibleObjectTests : IClassFixture<ThreadExceptionFixture>
+    {
+        [WinFormsFact]
+        public void CheckedListBoxItemAccessibleObject_Ctor_OwningCheckedListBoxIsNull_RaiseExeption()
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:14]:
```suggestion
        public void CheckedListBoxItemAccessibleObject_Ctor_OwningCheckedListBoxIsNull_RaiseException()
```

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+        {
+            using CheckedListBox checkedListBox = new();
+            var accessibleObject = (CheckedListBoxAccessibleObject)checkedListBox.AccessibilityObject;
+
+            Assert.Throws<ArgumentNullException>(() => { new CheckedListBoxItemAccessibleObject(null, new ItemArray.Entry("A"), accessibleObject); });
+            Assert.False(checkedListBox.IsHandleCreated);
+        }
+
+        [WinFormsFact]
+        public void CheckedListBoxItemAccessibleObject_Ctor_ItemCheckedListBoxIsNull_RaiseExeption()
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:24]:
```suggestion
        public void CheckedListBoxItemAccessibleObject_Ctor_ItemCheckedListBoxIsNull_RaiseException()
```

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+        {
+            using CheckedListBox checkedListBox = new();
+            var accessibleObject = (CheckedListBoxAccessibleObject)checkedListBox.AccessibilityObject;
+
+            Assert.Throws<ArgumentNullException>(() => { new CheckedListBoxItemAccessibleObject(checkedListBox, null, accessibleObject); });
+            Assert.False(checkedListBox.IsHandleCreated);
+        }
+
+        [WinFormsFact]
+        public void CheckedListBoxItemAccessibleObject_Ctor_ParentAccessibleObjectIsNull_RaiseExeption()
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:34]:
```suggestion
        public void CheckedListBoxItemAccessibleObject_Ctor_ParentAccessibleObjectIsNull_RaiseException()
```

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+            }
+
+            AccessibleObject accessibleObject = checkedListBox.AccessibilityObject;
+
+            for (int index = 0; index < 3; index++)
+            {
+                var itemAccessibleObject = (CheckedListBoxItemAccessibleObject)accessibleObject.GetChild(index);
+
+                Assert.Equal(index, itemAccessibleObject.TestAccessor().Dynamic.CurrentIndex);
+            }
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:82]:
```suggestion
            Assert.Equal(index, accessibleObject.GetChild(0).TestAccessor().Dynamic.CurrentIndex);
            Assert.Equal(index, accessibleObject.GetChild(1).TestAccessor().Dynamic.CurrentIndex);
            Assert.Equal(index, accessibleObject.GetChild(2).TestAccessor().Dynamic.CurrentIndex);
```

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+            AccessibleObject itemAccessibleObject3 = checkedListBox.AccessibilityObject.GetChild(2);
+
+            Assert.Null(itemAccessibleObject1.FragmentNavigate(UiaCore.NavigateDirection.PreviousSibling));
+            Assert.Equal(itemAccessibleObject1, itemAccessibleObject2.FragmentNavigate(UiaCore.NavigateDirection.PreviousSibling));
+            Assert.Equal(itemAccessibleObject2, itemAccessibleObject3.FragmentNavigate(UiaCore.NavigateDirection.PreviousSibling));
+            Assert.False(checkedListBox.IsHandleCreated);
+        }
+
+        [WinFormsFact]
+        public void CheckedListBoxItemAccessibleObject_DefaultAction_IfControlIsNotCreated_ReturnsEmptyString()
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:175]: Please fix it in the tests below
```suggestion
        public void CheckedListBoxItemAccessibleObject_DefaultAction_IfHandleIsNotCreated_ReturnsEmptyString()
```

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+        public void CheckedListBoxItemAccessibleObject_DoDefaultAction_IfControlIsCreated_WorksExpected(bool isChecked)
+        {
+            using CheckedListBox checkedListBox = new();
+            checkedListBox.Items.Add("A");
+            checkedListBox.SetItemChecked(0, isChecked);
+            checkedListBox.CreateControl();
+
+            checkedListBox.AccessibilityObject.GetChild(0).DoDefaultAction();
+
+            Assert.Equal(!isChecked, checkedListBox.GetItemChecked(0));
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:243]: Please add same check before calling `DoDefaultAction` method

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 14]
```diff
@@ -0,0 +1,260 @@
...
+        [InlineData(false)]
+        public void CheckedListBoxItemAccessibleObject_DoDefaultAction_IfControlIsNotCreated_DoNothing(bool isChecked)
+        {
+            using CheckedListBox checkedListBox = new();
+            checkedListBox.Items.Add("A");
+            checkedListBox.SetItemChecked(0, isChecked);
+
+            checkedListBox.AccessibilityObject.GetChild(0).DoDefaultAction();
+
+            Assert.Equal(isChecked, checkedListBox.GetItemChecked(0));
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:227]: Please add same check before calling `DoDefaultAction` method

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 15]
```diff
@@ -0,0 +1,260 @@
...
+        [InlineData(true, ((int)UiaCore.ToggleState.On))]
+        [InlineData(false, ((int)UiaCore.ToggleState.Off))]
+        public void CheckedListBoxItemAccessibleObject_ToggleState_IfControlIsCreated_ReturnsExpected(bool isChecked, int expected)
+        {
+            using CheckedListBox checkedListBox = new();
+            checkedListBox.Items.Add("A");
+            checkedListBox.SetItemChecked(0, isChecked);
+
+            Assert.Equal((UiaCore.ToggleState)expected, checkedListBox.AccessibilityObject.GetChild(0).ToggleState);
+            Assert.False(checkedListBox.IsHandleCreated);
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:257]: The test has a `CheckedListBoxItemAccessibleObject_ToggleState_IfControlIsCreated_ReturnsExpected` name, but the `Handle` is not created

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 16]
```diff
@@ -0,0 +1,260 @@
...
+            checkedListBox.SetItemChecked(0, isChecked);
+
+            Assert.Equal(isChecked.ToString(), checkedListBox.AccessibilityObject.GetChild(0).Value);
+            Assert.False(checkedListBox.IsHandleCreated);
+        }
+
+        [WinFormsTheory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void CheckedListBoxItemAccessibleObject_DoDefaultAction_IfControlIsNotCreated_DoNothing(bool isChecked)
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:219]: Please add the same test for the `Toggle` method

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs [hunk 17]
```diff
@@ -0,0 +1,260 @@
...
+            checkedListBox.AccessibilityObject.GetChild(0).DoDefaultAction();
+
+            Assert.Equal(isChecked, checkedListBox.GetItemChecked(0));
+            Assert.False(checkedListBox.IsHandleCreated);
+        }
+
+        [WinFormsTheory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void CheckedListBoxItemAccessibleObject_DoDefaultAction_IfControlIsCreated_WorksExpected(bool isChecked)
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxItemAccessibleObjectTests.cs:234]: Please add the same test for the `Toggle` method

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.cs [hunk 18]
```diff
@@ -845,6 +845,12 @@ protected override void OnKeyPress(KeyPressEventArgs e)
         protected virtual void OnItemCheck(ItemCheckEventArgs ice)
         {
             _onItemCheck?.Invoke(this, ice);
+
+            if (IsAccessibilityObjectCreated)
+            {
+                AccessibleObject checkedItem = AccessibilityObject.GetChild(ice.Index);
```
[the remark this answers is not in the document]
Comment (DmitryGorokhov) [reply src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.cs:851]: I consider that the `checkedItem` is not null here because we have to choose an item before checking and navigate to it so selected item has been already contained in the accessible objects inner collection of control accessible object.

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs (hunk 7 above)
Comment (DmitryGorokhov) [reply src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs:147]: In other test files I saw both ways so I think it is optional and I can choose one of them if it has the same performance

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs (hunk 5 above)
Comment (DmitryGorokhov) [reply src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs:42]: I'll check it

Comment (DmitryGorokhov) [reply src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs:42]: It changes because of this:
- Before this commit `CheckedListBoxAccessibleObject` class was based on `ControlAccessibleObject`  class and didn't override `ControlTypePropertyId` value. So the value filled from base implementation using the following "AccessibleRole-ControlType" mapping for `ControlAccessibleObject` (`Default` => `Pane`, you can see more details in docs).
- In this update `CheckedListBoxAccessibleObject` inherits `ListBoxAccessibleObject` and his implementation of this property. So for now this property has `ListControlTypeId` value by a `Default` role as a `ListBoxAccessibleObject` base class.
I don't know how it should exactly works if control is not created, but I think that this fix can be accepted. May be you or testers team will say that it is not correct and I'll change it
@SergeySmirnov-Akvelon, @vladimir-krestov, so, what do you think about it?

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxItemAccessibleObjectTests.cs [hunk 19]
```diff
@@ -0,0 +1,292 @@
...
+            Assert.Equal(testName, checkedListBox.AccessibilityObject.GetChild(0).Name);
+            Assert.False(checkedListBox.IsHandleCreated);
+        }
+
+        [WinFormsFact]
+        public void CheckedListBoxItemAccessibleObject_Role_IsCheckButton()
+        {
+            using CheckedListBox checkedListBox = new();
+            string testValue = "A";
+            checkedListBox.Items.Add(testValue);
```
Comment (SergeySmirnov-Akvelon) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxItemAccessibleObjectTests.cs:59]:
```suggestion
            checkedListBox.Items.Add("A");
```

**Comment by DmitryGorokhov** [comment]
Fix a test issues is in progress

**Comment by dreddy-work** [comment]
@DmitryGorokhov , once you are ready on this please notify CTI to validate this change specifically by providing the instructions.

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs (hunk 5 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/CheckedListBoxAccessibleObjectTests.cs:42]: ListControl looks like a better match.

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs [hunk 20]
```diff
@@ -2,120 +2,75 @@
...
+            private bool IsItemChecked => _owningCheckedListBox.GetItemChecked(CurrentIndex);
+
+            internal override bool IsPatternSupported(UiaCore.UIA patternId)
+                => patternId switch
                 {
-                    return AccessibleRole.CheckButton;
-                }
-            }
+                    UiaCore.UIA.InvokePatternId => true,
+                    UiaCore.UIA.LegacyIAccessiblePatternId => true,
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs:61]: LegacyIAccessible can be inherited from the base class. Line 61 can be removed.

File: src/System.Windows.Forms/tests/UnitTests/CheckedListBoxItemAccessibleObjectTests.cs [hunk 21]
```diff
@@ -0,0 +1,291 @@
...
+
+            Assert.Equal(0, accessibleObject.GetChild(0).TestAccessor().Dynamic.CurrentIndex);
+            Assert.Equal(1, accessibleObject.GetChild(1).TestAccessor().Dynamic.CurrentIndex);
+            Assert.Equal(2, accessibleObject.GetChild(2).TestAccessor().Dynamic.CurrentIndex);
+
+            Assert.False(checkedListBox.IsHandleCreated);
+        }
+
+        [WinFormsTheory]
+        [InlineData(((int)UiaCore.UIA.TogglePatternId))]
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/CheckedListBoxItemAccessibleObjectTests.cs:83]: Can parentheses be simplified?

Comment (DmitryGorokhov) [reply src/System.Windows.Forms/tests/UnitTests/CheckedListBoxItemAccessibleObjectTests.cs]: Sure. Such cases have also been reworked, thanks

**Review by dreddy-work — DISMISSED** [review]

File: src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/MultipleControls.resx [hunk 22]
```diff
@@ -1,5 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/MultipleControls.resx:1]: Do you have any reasons why did you change it? If no, return it as was.

**Review by vladimir-krestov — APPROVED** [review]

File: src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/MultipleControls.resx (hunk 22 above)
Comment (DmitryGorokhov) [reply src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/MultipleControls.resx]: No, I have updates in designer only. This file has been restored

**Comment by DmitryGorokhov** [comment]
Remind: fix a test issues are still in progress...

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs [hunk 23]
```diff
@@ -2,129 +2,88 @@
...
                 {
-                    if (!ParentCheckedListBox.IsHandleCreated)
+                    if (!_owningCheckedListBox.IsHandleCreated)
                     {
                         return AccessibleStates.None;
                     }
 
                     AccessibleStates state = AccessibleStates.Selectable | AccessibleStates.Focusable;
 
+                    if (!_owningAccessibleObject.BoundingRectangle.IntersectsWith(Bounds))
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs:79]:
```suggestion
                    if (!Parent.BoundingRectangle.IntersectsWith(Bounds))
```

File: src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ItemAccessibleObject.cs [hunk 24]
```diff
@@ -17,7 +17,7 @@ public partial class ListBox
         internal class ListBoxItemAccessibleObject : AccessibleObject
         {
             private readonly ItemArray.Entry _itemEntry;
-            private readonly ListBoxAccessibleObject _owningAccessibleObject;
+            private protected readonly ListBoxAccessibleObject _owningAccessibleObject;
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ItemAccessibleObject.cs:20]: remove `protected`
```suggestion
            private readonly ListBoxAccessibleObject _owningAccessibleObject;
```

**Comment by DmitryGorokhov** [comment]
Actually, both problems are fixed. Waiting for test results

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs [hunk 25]
```diff
@@ -2,129 +2,88 @@
...
+            private bool IsItemChecked => _owningCheckedListBox.GetItemChecked(CurrentIndex);
+
+            internal override bool IsPatternSupported(UiaCore.UIA patternId)
+                => patternId switch
                 {
-                    return AccessibleRole.CheckButton;
-                }
-            }
+                    UiaCore.UIA.InvokePatternId => true,
+                    UiaCore.UIA.TogglePatternId => true,
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs:61]: If the control is disabled, should toggle pattern be not available?

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs [hunk 26]
```diff
@@ -2,129 +2,88 @@
...
 
                     AccessibleStates state = AccessibleStates.Selectable | AccessibleStates.Focusable;
 
+                    if (!Parent.BoundingRectangle.IntersectsWith(Bounds))
+                    {
+                        state |= AccessibleStates.Offscreen;
+                    }
+
                     // Checked state
                     //
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs:85]: nit: remove the empty comment line

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs (hunk 25 above)
Comment (DmitryGorokhov) [reply src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs:61]: No, because if don't support toggle pattern we receive the error in Accessibility Insights test as you can see on screen.

image

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs (hunk 26 above)
Comment (DmitryGorokhov) [reply src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs]: in both places

**Comment by DmitryGorokhov** [comment]
I have 2 from 3 fixed test-issue. One problem aren't resolved. So, PR is still WIP

**Comment by DmitryGorokhov** [comment]
Testers approved the fix: there are no regression issues here.

File: src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs (hunk 25 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.CheckedListBoxItemAccessibleObject.cs:61]: Makes sense, thank you for the investigation.

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good

**Comment by Tanya-Solyanik** [comment]
@DmitryGorokhov - are you `ready to merge`?

**Review by RussKie — APPROVED** [review]

**Comment by Philip-Wang01** [comment]
Verified this issue with .NET SDK 7.0.100-alpha.1.21479.33, Add support UIA provider for CheckedListBox control
image
image
image
